src/migrate_411_412.ml

```diff
@@ -0,0 +1,1204 @@
+  =
+  fun
+    { Ast_411.Parsetree.pval_name = pval_name;
+      Ast_411.Parsetree.pval_type = pval_type;
+      Ast_411.Parsetree.pval_prim = pval_prim;
+      Ast_411.Parsetree.pval_attributes = pval_attributes;
+      Ast_411.Parsetree.pval_loc = pval_loc }
+    ->
+    {
+      Ast_412.Parsetree.pval_name = (copy_loc (fun x -> x) pval_name);
+      Ast_412.Parsetree.pval_type = (copy_core_type pval_type);
+      Ast_412.Parsetree.pval_prim = (List.map (fun x -> x) pval_prim);
+      Ast_412.Parsetree.pval_attributes = (copy_attributes pval_attributes);
+      Ast_412.Parsetree.pval_loc = (copy_location pval_loc)
+    }
+and copy_object_field_desc :
+  Ast_411.Parsetree.object_field_desc -> Ast_412.Parsetree.object_field_desc
+  =
+  function
+  | Ast_411.Parsetree.Otag (x0, x1) ->
+      Ast_412.Parsetree.Otag ((copy_loc copy_label x0), (copy_core_type x1))
+  | Ast_411.Parsetree.Oinherit x0 ->
+      Ast_412.Parsetree.Oinherit (copy_core_type x0)
+and copy_arg_label : Ast_411.Asttypes.arg_label -> Ast_412.Asttypes.arg_label
+  =
+  function
+  | Ast_411.Asttypes.Nolabel -> Ast_412.Asttypes.Nolabel
+  | Ast_411.Asttypes.Labelled x0 -> Ast_412.Asttypes.Labelled x0
+  | Ast_411.Asttypes.Optional x0 -> Ast_412.Asttypes.Optional x0
+and copy_closed_flag :
+  Ast_411.Asttypes.closed_flag -> Ast_412.Asttypes.closed_flag =
+  function
+  | Ast_411.Asttypes.Closed -> Ast_412.Asttypes.Closed
+  | Ast_411.Asttypes.Open -> Ast_412.Asttypes.Open
+and copy_label : Ast_411.Asttypes.label -> Ast_412.Asttypes.label =
+  fun x -> x
+and copy_rec_flag : Ast_411.Asttypes.rec_flag -> Ast_412.Asttypes.rec_flag =
+  function
+  | Ast_411.Asttypes.Nonrecursive -> Ast_412.Asttypes.Nonrecursive
+  | Ast_411.Asttypes.Recursive -> Ast_412.Asttypes.Recursive
+and copy_constant : Ast_411.Parsetree.constant -> Ast_412.Parsetree.constant
+  =
+  function
+  | Ast_411.Parsetree.Pconst_integer (x0, x1) ->
+      Ast_412.Parsetree.Pconst_integer (x0, (Option.map (fun x -> x) x1))
+  | Ast_411.Parsetree.Pconst_char x0 -> Ast_412.Parsetree.Pconst_char x0
+  | Ast_411.Parsetree.Pconst_string (x0, x1, x2) ->
+      Ast_412.Parsetree.Pconst_string
+        (x0, (copy_location x1), (Option.map (fun x -> x) x2))
+  | Ast_411.Parsetree.Pconst_float (x0, x1) ->
+      Ast_412.Parsetree.Pconst_float (x0, (Option.map (fun x -> x) x1))
+and copy_Longident_t : Longident.t -> Longident.t =
+  function
+  | Longident.Lident x0 -> Longident.Lident x0
+  | Longident.Ldot (x0, x1) -> Longident.Ldot ((copy_Longident_t x0), x1)
+  | Longident.Lapply (x0, x1) ->
+      Longident.Lapply ((copy_Longident_t x0), (copy_Longident_t x1))
+and copy_loc :
+  'f0 'g0 .
+    ('f0 -> 'g0) -> 'f0 Ast_411.Asttypes.loc -> 'g0 Ast_412.Asttypes.loc
+  =
+  fun f0 ->
+    fun { Ast_411.Asttypes.txt = txt; Ast_411.Asttypes.loc = loc } ->
+      {
+        Ast_412.Asttypes.txt = (f0 txt);
+        Ast_412.Asttypes.loc = (copy_location loc)
+      }
+and copy_location : Location.t -> Location.t =
```

Comment:
I think this one can also be turned into identity, as probably can all the ones that were in the opposite direction!